### PR TITLE
Support Mixtral quantization using INC

### PR DESCRIPTION
**Pull request by dudilester**

### CODE REVIEW

File: vllm/hpu/ops.py [hunk 1]
```diff
@@ -148,3 +117,56 @@ def prompt_attention(
         attn_weights = attn_weights.flatten(1, 2)
     attn_weights = attn_weights.transpose(1, 2)
     return attn_weights
+
+
+class MoeMatmul(torch.nn.Module):
```
Comment (Tiefen-boop) [inline vllm/hpu/ops.py:122]: How do you benefit it being a `torch.nn.Module` implementation?

Comment (Yantom1) [reply vllm/hpu/ops.py:122]: That way we can patch it in INC for quantization

File: vllm/hpu/ops.py [hunk 2]
```diff
@@ -265,4 +234,61 @@ def dispatch_bgmv_embedding(
...
+        htorch.core.mark_step()
+
+        for expert_idx in range(self.num_total_experts):
+            padded_weight = padded_weights[expert_idx]
+            w_output = self.w13_list[expert_idx].calc(hidden_states,
+                                                      expert_idx, w1)
+            w_output = silu_and_mul(w_output)
+            w_output = self.w2_list[expert_idx].calc(w_output, expert_idx, w2)
+            final_hidden_states += w_output * padded_weight
+            htorch.core.mark_step()
```
Comment (szutenberg) [inline vllm/hpu/ops.py:292]: Not needed. See https://github.com/HabanaAI/vllm-fork/pull/231

File: vllm/worker/habana_model_runner.py [hunk 3]
```diff
@@ -1680,9 +1680,7 @@ def shutdown_inc(self):
         if (model_config := getattr(self, "model_config", None)) and \
                          getattr(model_config, "quantization", None) == 'inc':
             print('inc shutdown start')
-            from neural_compressor.torch.quantization import (
-                finalize_calibration)
-            finalize_calibration(self.model.model)
+            #finalize_calibration(self.model.model)
```
Comment (szutenberg) [inline vllm/worker/habana_model_runner.py:1683]: Why?

File: vllm/model_executor/layers/fused_moe/layer.py [hunk 4]
```diff
@@ -13,9 +13,6 @@
 from vllm.model_executor.utils import set_weight_attrs
```
Comment (szutenberg) [inline vllm/model_executor/layers/fused_moe/layer.py:13]: @kzawora-intel could you check if these changes are upstreamable?

**Review by szutenberg — CHANGES_REQUESTED** [review]

File: vllm/hpu/ops.py (hunk 2 above)
Comment (Tiefen-boop) [reply vllm/hpu/ops.py]: Removed

File: vllm/worker/habana_model_runner.py (hunk 3 above)
Comment (Tiefen-boop) [reply vllm/worker/habana_model_runner.py]: reverted

File: vllm/worker/habana_model_runner.py [hunk 5]
```diff
@@ -16,6 +16,7 @@
 
 import habana_frameworks.torch as htorch
 import torch
+from neural_compressor.torch.quantization import finalize_calibration
```
Comment (MrGeva) [inline vllm/worker/habana_model_runner.py:19]: revert the changes in this file, they are not needed.

**Review by MrGeva — APPROVED** [review]
LGTM :)

**Review by szutenberg — APPROVED** [review]

File: vllm/worker/habana_model_runner.py (hunk 5 above)
Comment (jkaniecki) [inline vllm/worker/habana_model_runner.py:19]: This brakes taking fp8 measurements with TP>1 - please revert changes made to this file

**Review by jkaniecki — CHANGES_REQUESTED** [review]
